metrics/web/docs/features/navigation/diagrams/internal_navigation_sequence_diagram.puml

```diff
@@ -0,0 +1,48 @@
+@startuml internal_navigation_sequence_diagram
+skinparam ParticipantPadding 20
+skinparam BoxPadding 10
+
+actor Bob
+
+
+
+box Managing Pages
+participant NavigationNotifier
+participant MetricsPageFactory
+participant MetricsRouterDelegate
+endbox
+
+participant MetricsRouteInformationParser
+
+participant Router
+
+Bob -> NavigationNotifier : pushNamed('newPage')
+activate NavigationNotifier
+
+NavigationNotifier -> MetricsPageFactory : create(configuration)
+activate MetricsPageFactory
+
+return updated metricsPage
+
+NavigationNotifier -> NavigationNotifier : add metricsPage to the pages
+
+NavigationNotifier -> MetricsRouterDelegate : notifyListeners()
+deactivate NavigationNotifier
+activate MetricsRouterDelegate
+
+MetricsRouterDelegate -> MetricsRouterDelegate: applies the updated pages
+
+MetricsRouterDelegate -> MetricsRouteInformationParser : restoreRouteInformation(configuration)
+activate MetricsRouteInformationParser
+
+MetricsRouterDelegate --> Bob : shows a new metricsPage
+deactivate MetricsRouterDelegate
+
+MetricsRouteInformationParser -> Router : notifies about page changed
```

Comment:
Does it notify `Router` or `RouteProvider`? In another sequence diagram, the provider interacts directly with the parser, and the parser with the delegate - this seems to be a mistake. AFAIK, the Router itself manages interactions between provider, delegate, parser (as the Router is the only class that has all of them in one class). So there must not be a parser that calls delegate as the parser has no delegate as a member.